utils/cautils/token_generator.go

```diff
@@ -161,32 +162,68 @@ func generateK8sSAToken(ctx *cli.Context, p *provisioner.K8sSA) (string, error)
 func generateX5CToken(ctx *cli.Context, p *provisioner.X5C, tokType int, tokAttrs tokenAttrs) (string, error) {
 	x5cCertFile := ctx.String("x5c-cert")
 	x5cKeyFile := ctx.String("x5c-key")
+	x5cChainFiles := ctx.StringSlice("x5c-chain")
+	kmsURI := ctx.String("kms")
+
 	if x5cCertFile == "" {
 		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-cert")
 	}
 	if x5cKeyFile == "" {
 		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-key")
 	}
 
-	// Get private key from given key file
-	var opts []jose.Option
-	if passOpt := getProvisionerPasswordOption(ctx); passOpt != nil {
-		opts = append(opts, passOpt)
-	}
-	jwk, err := jose.ReadKey(x5cKeyFile, opts...)
-	if err != nil {
-		return "", err
-	}
+	var jwk *jose.JSONWebKey
+	var err error
 
+	if kmsURI != "" {
+		var opts []pemutil.Options
+		var kmsSigner crypto.Signer
+		kmsSigner, err = cryptoutil.CreateSigner(ctx.String("kms"), x5cKeyFile, opts...)
+		if err != nil {
+			return "", err
+		}
+
+		joseSigner := jose.NewOpaqueSigner(&kmsSigner)
+
+		jwk = &jose.JSONWebKey{
+			Key:       joseSigner,
+			KeyID:     x5cKeyFile,
+			Algorithm: string(joseSigner.Algs()[0]),
+		}
+	} else {
+		// Get private key from given key file
+		var opts []jose.Option
+		if passOpt := getProvisionerPasswordOption(ctx); passOpt != nil {
+			opts = append(opts, passOpt)
+		}
+		jwk, err = jose.ReadKey(x5cKeyFile, opts...)
+		if err != nil {
+			return "", err
+		}
```

Comment:
You can do this with `cryptoutil.CreateSigner` as it is, even if that `jose` was used here, the key should be a PEM file.

`cryptoutil.CreateSigner` will use `pemutil.Read()` when `kms == ""` you will need to the `pemutil.Options` using a copy of the `getProvisionerPasswordOption()` function, but returning `pemutil.Options`

Reply:
Yes, but `keyutil.VerifyPair()` will verify the pair comparing the public keys, so at the end is the same.